### PR TITLE
fix(Core/Creatues): Fixed possible exploit where members would not attack if they're already in evade + added additional calls in DamageDeal calls

### PR DESCRIPTION
**Pull request by sanctum32**
This will address issue where member would not assist attacked member while they're in evade mode.
To reproduce this, use hunter or rogue like class and simply use any formation group which has 515 (or 512) flag and do these steps:
1) Pull any npc from formation
2) Let them to pass half way till you
3) Use "Feign Death" (hunters or just add spell to your spell book)
4) Once they begins moving back to home position, quickly deal damage to same target
5) Here we go, from several enemies, now we have only 1x enemy extracted away from others

This PR simply will remove evade state from returning members and will force members to assist formation group member again.
Currently if formation group members are in evade mode, they simply ignores member (member which is still has actual combat fight) calls and member is left extracted from group.

## Changes Proposed:
-  Adds additional MemberAttackStart calls in Unit::DamageDeal function (partially based on TrinityCore code)
-  Adds evade mode removal to formation group members which maybe are already returning (in evade mode) and members will can assist member in combat again

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1) Pull it
2) Let them to pass half way till you
3) Use "Feign Death"
4) Once they begins moving back to home position, quickly deal damage to same target
5) Here we go, from several enemies, now we have only 1x enemy extracted away from others

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -743,6 +743,24 @@ uint32 Unit::DealDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage
...
+        {
+            if (CreatureGroup* group = pCreature->GetFormation())
+                group->MemberAttackStart(pCreature, victim, true);
+        }
+
+        // Trigger victim unit formation
+        if (Creature* pvictimCreat = victim->ToCreature())
+        {
+            if (CreatureGroup* group = pvictimCreat->GetFormation())
+                group->MemberAttackStart(pvictimCreat, attacker, true);
```
[the remark this answers is not in the document]
Comment (temperrr) [reply src/server/game/Entities/Unit/Unit.cpp:760]: Could you give me some examples of monsters with this specific flag? (I was unable to do it without PR on Boss Tho'Grun in Badlands for example, he does not have this flag.)

**Comment by temperrr** [comment]
Could you give me some examples of monsters with this specific flag? (I was unable to do it without PR on Boss Tho'Grun in Badlands for example, he does not have this flag.)

**Comment by Nefertumm** [comment]
Solve merge conflicts please.

**Comment by Winfidonarleyan** [comment]
@sanctum32 Please give examples so that testers can test this properly

**Pull request closed without merging**